test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1

```diff
@@ -135,49 +135,85 @@ Describe "Import-Module for Binary Modules in GAC" -Tags 'Feature' {
 
 Describe "Import-Module for Binary Modules" -Tags 'CI' {
 
-    It "PS should try to load the assembly from file path first" {
- $src = @"
-using System.Management.Automation;           // Windows PowerShell namespace.
-
-namespace ModuleCmdlets
-{
-  [Cmdlet(VerbsDiagnostic.Test,"BinaryModuleCmdlet1")]
-  public class TestBinaryModuleCmdlet1Command : Cmdlet
-  {
-    protected override void BeginProcessing()
-    {
-      WriteObject("BinaryModuleCmdlet1 exported by the ModuleCmdlets module.");
-    }
-  }
-}
+    BeforeAll {
+        $src = @"
+            using System.Management.Automation;
+            namespace ModuleCmdlets
+            {
+                [Cmdlet(VerbsDiagnostic.Test,"BinaryModuleCmdlet1")]
+                public class TestBinaryModuleCmdlet1Command : Cmdlet
+                {
+                    protected override void BeginProcessing()
+                    {
+                        WriteObject("BinaryModuleCmdlet1 exported by the ModuleCmdlets module.");
+                    }
+                }
+            }
 "@
+        Add-Type -TypeDefinition $src -OutputAssembly $TESTDRIVE\System.dll
+        Add-Type -TypeDefinition $src -OutputAssembly $TESTDRIVE\System.exe
+    }
 
-    Add-Type -TypeDefinition $src -OutputAssembly $TESTDRIVE\System.dll
-    $results = pwsh -noprofile -c "`$module = Import-Module $TESTDRIVE\System.dll -Passthru; `$module.ImplementingAssembly.Location; Test-BinaryModuleCmdlet1"
+    It "PS should try to load the binary module from a file path with extension: <extension>" -TestCases (
+        @{extension = "dll"},
+        @{extension = "exe"}
+     ) {
+         param ($extension)
 
-    #Ignore slash format difference under windows/Unix
-    $path = (Get-ChildItem $TESTDRIVE\System.dll).FullName
-    $results[0] | Should -BeExactly $path
-    $results[1] | Should -BeExactly "BinaryModuleCmdlet1 exported by the ModuleCmdlets module."
+        $TestModulePath = Join-Path $TESTDRIVE "System.$extension"
+        $assemblyLocation, $cmdletOutput = Start-Job -ScriptBlock {
+            $module = Import-Module $using:TestModulePath -Passthru;
+            $module.ImplementingAssembly.Location;
+            Test-BinaryModuleCmdlet1
+        } | Wait-Job -Timeout 30 | Receive-Job
```

Comment:
Can we make sure we remove the job before the tests are done?

In the other new cases as well.

Reply:
Yes, will add